crates/npmrc/src/custom_deserializer.rs

```diff
@@ -0,0 +1,72 @@
+use std::{env, path::Path, str::FromStr};
+
+use serde::{de, Deserialize, Deserializer};
+
+// This needs to be implemented because serde doesn't support default = "true" as
+// a valid option, and throws  "failed to parse" error.
+pub fn bool_true() -> bool {
+    true
+}
+
+pub fn default_hoist_pattern() -> Vec<String> {
+    vec!["*".to_string()]
+}
+
+pub fn default_public_hoist_pattern() -> Vec<String> {
+    vec!["*eslint*".to_string(), "*prettier*".to_string()]
+}
+
+/// If the $PACQUET_HOME env variable is set, then $PACQUET_HOME/store
+/// If the $XDG_DATA_HOME env variable is set, then $XDG_DATA_HOME/pacquet/store
+/// On Windows: ~/AppData/Local/pacquet/store
+/// On macOS: ~/Library/pacquet/store
+/// On Linux: ~/.local/share/pacquet/store
```

Comment:
You may want to consider https://crates.io/crates/directories

Here is like, how complicated this problem is https://poignardazur.github.io/2023/05/23/platform-compliance-in-cargo/

Reply:
I agree, and I don't like this code as well, but I wanted to be 100% in sync with pnpm's defaults. Ref: https://pnpm.io/npmrc#store-dir

Reply:
ahhh nice